feat: add prettier

henryiii opened the pull request:
Should be implemented in Cookie first.

eduardo-rodrigues commented:
How does this interfere, if at all, with for example Black? This point may be worth a sentence.

henryiii commented:
Pretty sure Python is not one of the languages it supports. But that's partially why it has a mention of using `types_or` to ensure it's not trying to change files you'd rather not be auto formatted. I should include json in the list. I'm pretty sure we don't already recommend an auto formatter for any of the supported file types.

<img width="295" alt="Screen Shot 2022-05-05 at 4 46 31 PM" src="https://user-images.githubusercontent.com/4616906/167022985-df820df4-9a43-451f-8bb2-7b14e73cc68f.png">

henryiii commented on pages/developers/style.md:
````diff
@@ -532,6 +532,26 @@ If you have shell scripts, you can protect against common mistakes using [shellc
+```
+
+Since this formats a variety of very common file types (like `html`, `md`,
+`yaml`, `js`, and more), you will usually want to provide a `types_or` setting
````
```suggestion
`yaml`, `js`, `json`, and more), you will usually want to provide a `types_or` setting
```

henryiii commented on pages/developers/style.md:
```diff
@@ -532,6 +532,26 @@ If you have shell scripts, you can protect against common mistakes using [shellc
+    rev: "v2.6.2"
+    hooks:
+      - id: prettier
+        types_or: [yaml, markdown, html, js]
```
```suggestion
        types_or: [yaml, markdown, html, js, json]
```

eduardo-rodrigues replied:
Got it. Thanks.

henryiii commented:
Updates: I've run prettier on the website itself, and worked out a few things:

* Prettier can format YAML inside codeblocks in markdown! That's great, so I added blacken-docs which does the same thing for Python (Black) inside markdown.
    * This actually found a bug, we had invalid syntax in one example!
* I'm having prettier reflow everything to ~80 chars. We could also have it reflow to no line wrap, I care more about consistency than which setting.
* Pretty _mostly_ is okay with Jekyll, but it does break `{: .no_toc}` or similar right after a heading by adding a blank line. I've avoided this most places by adding the title directly to the page - you can override the title with `custom_title: <>`. If you add `custom_title` but leave it blank, there will be no title at the top.
* I've replaced the ugly capture workarounds with the correct `markdown="1"` for expandable. Later I'll style the expandable to look more like the current TOC styling.
* There was some slightly broken or non standard YAML that I've cleaned up with prettier's help.

eduardo-rodrigues reviewed (approved):
Difficult to navigate the many tiny changes but all seems fine.
